worlds/overcooked2/Logic.py

```diff
@@ -0,0 +1,3898 @@
+from BaseClasses import CollectionState
+from .Overcooked2Levels import Overcooked2GenericLevel, Overcooked2Dlc, Overcooked2Level
+from typing import Dict
+from random import Random
+
+
+def has_requirements_for_level_access(state: CollectionState, level_name: str, previous_level_completed_event_name: str,
+                                      required_star_count: int, player: int) -> bool:
```

Comment:
the function that take CollectionState as first argument could be done using logic mixin. Not saying we *have to* do that, but I think other games do it - so instead of
`lambda state: has_requirements_for_level_access(state, ...)` you end up having
`lambda state: state.oc2_has_requirements_for_level_access(...)`
and instead of `state.something`, you do `self.something` in the body.

Reply:
"other games do it" is how I got into trouble in several of the previous feedback points. What is the tangible benefit of replacing my already vetted implementation with a different one?